code-challenges.test.js

```diff
@@ -17,35 +17,166 @@
+
+const newVotes = (number) =>{
+  const {upVotes: 13, downVotes: 2}
+}
+console.log(newVotes(votes1))
+
+
+
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
-const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
-// Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+// const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
+// const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
+// // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+
+// describe("newArray", () => {
+//   it("create a function that takes two arrays and returns one with no duplicate values", () => {
+//     expect(newArray(dataTypesArray1)).toEqual(["array", "object", "number", "string", "Boolean"])
+//     expect(newArray(dataTypesArray2)).toEqual(["string", "null", "Boolean", "string", "undefined"])
+//   })
+// })
+
+//RED - Good Failure
+// FAIL  ./code-challenges.test.js
+// newArray
+//   ✕ create a function that takes an object that contains up & down votes and returns the total tally
+
+// ● newArray › create a function that takes an object that contains up & down votes and returns the total tally
+
+//   ReferenceError: newArray is not defined
+
+//     116 | describe("newArray", () => {
+//     117 |   it("create a function that takes two arrays and returns one with no duplicate values", () => {
+//   > 118 |     expect(newArray(dataTypesArray1)).toEqual(["array", "object", "number", "string", "Boolean"])
+//         |     ^
+//     119 |     expect(newArray(dataTypesArray2)).toEqual(["string", "null", "Boolean", "string", "undefined"])
+//     120 |   })
+//     121 | })
+
+//     at Object.expect (code-challenges.test.js:118:5)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+// Snapshots:   0 total
+// Time:        0.161 s, estimated 1 s
+// Ran all test suites.
+// error Command failed wi
 
 // b) Create the function that makes the test pass.
+//PseudoCode
+//Input: dataTypesArray1, dataTypesArray2
+//Output: ["array", "object", "number", "string", "Boolean", "null", "undefined"], both arrays combined, without repeating values
+//Process: Create a function called newArray that takes in two arrays as arguments. Combine both arrays using the object 'Set'. Was confused on this one so I was trying to figure out this througth notes an ChatGPT.
```

Comment:
Good effort trying to use Set, but make sure you are fully understanding why it is being used